flink-runtime/src/main/java/org/apache/flink/runtime/instance/SlotSharingGroupAssignment.java

```diff
@@ -576,7 +560,7 @@ private void internalDisposeEmptySharedSlot(SharedSlot sharedSlot) {
 		
 		if (parent == null) {
 			// root slot, return to the instance.
-			sharedSlot.getInstance().returnAllocatedSlot(sharedSlot);
+			sharedSlot.getOwner().returnAllocatedSlot(sharedSlot);
```

Comment:
Can we use `sharedSlot.releaseSlot` instead? It seems not so right to get the owner which is then used to call `returnAllocatedSlot` with the same slot. I think this violates the law of Demeter.

Alternatively, we could add a method `returnSlot` to `Slot` which does: `owner.returnAllocatedSlot(this)`.